bootstrap.py

```
@@ -0,0 +1,256 @@
+        "text": True,
+    }
+    if log_file:
+        log_file.parent.mkdir(parents=True, exist_ok=True)
+        handle = log_file.open("w", encoding="utf-8")
+        kwargs["stdout"] = handle
+        kwargs["stderr"] = subprocess.STDOUT
+    else:
+        kwargs["stdout"] = None
+        kwargs["stderr"] = None
+    try:
+        result = subprocess.run(resolved, **kwargs)
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)